src/idrs.jl

```diff
@@ -144,41 +152,56 @@ function idrs_core!{T}(X, op, args, C::T, s, tol, maxiter)
             # New column of M = P'*G  (first k-1 entries are zero)
 
             for i in k:s
-                M[i,k] = vecdot(P[i],G[k])       
+                M[i,k] = vecdot(P[i],G[k])
             end
 
-            #  Make r orthogonal to q_i, i = 1..k 
+            #  Make r orthogonal to q_i, i = 1..k
 
             beta = f[k]/M[k,k]
             axpy!(-beta, G[k], R)
             axpy!(beta, U[k], X)
-        
+
             normR = vecnorm(R)
+            if smoothing
+                copy!(T_s, R_s); axpy!(-1., R, T_s) # T_s = R_s - R
+                gamma = vecdot(R_s, T_s)/vecnorm(T_s)^2
+                axpy!(-gamma, T_s, R_s) # R_s = R_s - gamma*T_s

```

Comment:
This is very similar to the omega function, maybe a better abstraction? @astudillor